src/material.cpp

```diff
@@ -1483,7 +1483,9 @@ std::map<int, double> pyne::Material::get_density_frac(std::string frac_type){
     fracs = comp;
     if (density != -1.0) {
       for (comp_iter ci = fracs.begin(); ci != fracs.end(); ci++){
-        ci->second *= density;
+        if (mult_den) {
```

Comment:
Move this conditional up 2 lines.... no point in entering the loop or checking the value of `density` if this is not true

Reply:
Don't know why I put the conditional inside for loop...
The above comment makes sense except that
I assume you meant `if (density != -1.0 && mult_den):` instead of `if (density != -1.0 && !mult_den):`